### PR TITLE
introduce useWorldBySlug hook for world searching

### PR DESCRIPTION
**Pull request by denisdimitrov**
Adds a new hook for world searching by `worldSlug`. 

This hook will be required for the new in venue url `/in/:worldSlug/:venueSlug`

### CODE REVIEW

File: src/hooks/worlds/useWorldBySlug.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    {
+      initialData: undefined,
+    }
+  );
+
+  const isWorldLoaded = status !== "loading";
+
+  if (!worlds?.[0]) {
+    Bugsnag.notify(
+      `World with the following slug: ${worldSlug} does not exist.`,
```
Comment (colinhowe) [inline src/hooks/worlds/useWorldBySlug.ts:41]: I think we're going to see this a lot with bots crawling the site etc. Might be worth just throwing an error that eventually results in a 404

**Review by colinhowe — APPROVED** [review]
Looks good. Just a comment around error handling

**Comment by goran-peoski-work** [comment]
I think this is a good opportunity to rename the other one `useWorldById()`, makes it easy for other PR's to just merge 2 lines change in 15 files because of rename.

File: src/hooks/worlds/useWorldBySlug.ts (hunk 1 above)
Comment (goran-peoski-work) [reply src/hooks/worlds/useWorldBySlug.ts:41]: My initial reaction was mostly about spamming our Bugsnag, but I kind of let it go to first check if that causes an issue, I mean, the solution would be a simple replacement of `Bugsnag.notify()` with `console.error()` just to log it somewhere.

I don't think relying on errors unwinding stack should be the normal behavior though, but an exception to handle in a nicer way.

**Comment by denisdimitrov** [comment]
@colinhowe The error is now thrown 

https://github.com/sparkletown/sparkle/pull/2543/commits/ae1cb4f71c7508b74045778ed2404e4357611e4f

@goran-peoski-work The other hook is now called `useWorldById`


https://github.com/sparkletown/sparkle/pull/2543/commits/6ef584e767fb064eeba20941eb2abc1d60495137

**Pull request merged**